### PR TITLE
Updates to API v2 CORS and related for better browser integration

### PR DESCRIPTION
**Pull request by dhalperi**
This change is [Reviewable](https://reviewable.io/reviews/batfish/batfish/666)

### CODE REVIEW

File: projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
+import javax.ws.rs.core.Response;
 import javax.ws.rs.ext.Provider;
 
 /** Allow the system to serve xhr level 2 from all cross domain site */
 @Provider
-public class CrossDomainFilter implements ContainerResponseFilter {
+@Priority(-1) /* Highest priority to set the CORS filter as the first one to run. */
+public class CrossDomainFilter implements ContainerRequestFilter, ContainerResponseFilter {
+  /**
+   * Recognizes a CORS preflight request, and return OK without any further downstring processing in
```
Comment (haverma) [inline projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java]: Was "downstring" supposed to be "downstream" ?

File: projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
+   * such a case.
+   */
+  @Override
+  public void filter(ContainerRequestContext creq) {
+    if (HttpMethod.OPTIONS.equals(creq.getMethod())) {
+      // Cache these responses.
+      CacheControl control = new CacheControl();
+      control.setMaxAge(3600);
+      creq.abortWith(Response.ok().cacheControl(control).build());
```
Comment (haverma) [inline projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java]: Won't we need a header "Access-Control-Allow-Methods" in the response, listing the supported HTTP methods, apart from the OK response ? [https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request](reference)

**Comment by haverma** [comment]
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java, line 19 at r1](https://reviewable.io:443/reviews/batfish/batfish/666#-KyolGG4oBrpXzahiVly:-KyolGG4oBrpXzahiVlz:b2koc8p) ([raw file](https://github.com/batfish/batfish/blob/81f5a9b69892856176db02b0b436e5d212e6f629/projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java#L19)):*
> ```Java
> public class CrossDomainFilter implements ContainerRequestFilter, ContainerResponseFilter {
>   /**
>    * Recognizes a CORS preflight request, and return OK without any further downstring processing in
> ```

Was "downstring" supposed to be "downstream" ?

---

*[projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java, line 28 at r1](https://reviewable.io:443/reviews/batfish/batfish/666#-KyolOyI3Cnj1Iyp7YuI:-KyolOyI3Cnj1Iyp7YuJ:bknn3r) ([raw file](https://github.com/batfish/batfish/blob/81f5a9b69892856176db02b0b436e5d212e6f629/projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java#L28)):*
> ```Java
>       CacheControl control = new CacheControl();
>       control.setMaxAge(3600);
>       creq.abortWith(Response.ok().cacheControl(control).build());
> ```

Won't we need a header "Access-Control-Allow-Methods" in the response, listing the supported HTTP methods, apart from the OK response ? https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/666)*

**Comment by dhalperi** [comment]
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java, line 19 at r1](https://reviewable.io:443/reviews/batfish/batfish/666#-KyolGG4oBrpXzahiVly:-KyqN8jEeJ5h7etmWwol:b-896fix) ([raw file](https://github.com/batfish/batfish/blob/81f5a9b69892856176db02b0b436e5d212e6f629/projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java#L19)):*
<details><summary><i>Previously, haverma (Harsh Verma) wrote…</i></summary><blockquote>

Was "downstring" supposed to be "downstream" ?
</blockquote></details>

Done.

---

*[projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java, line 28 at r1](https://reviewable.io:443/reviews/batfish/batfish/666#-KyolOyI3Cnj1Iyp7YuI:-KyqMlIVD2k0IIFdpnPC:bwwm0my) ([raw file](https://github.com/batfish/batfish/blob/81f5a9b69892856176db02b0b436e5d212e6f629/projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java#L28)):*
<details><summary><i>Previously, haverma (Harsh Verma) wrote…</i></summary><blockquote>

Won't we need a header "Access-Control-Allow-Methods" in the response, listing the supported HTTP methods, apart from the OK response ? https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request
</blockquote></details>

That's handled by the response filter below. Both of them run on all requests: the request filter on the way in and the response filter on the way out.

The purpose of this code is to abort the "way in" path early.

I also removed the caching here as it is not necessary – it was the wrong type of caching; as that's handled by the `Access-Control-Max-Age` field below.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/666)*

**Comment by haverma** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/666#-:-Kyqcq03xp45vK1FIMG9:bnfp4nl)*

**Pull request merged**